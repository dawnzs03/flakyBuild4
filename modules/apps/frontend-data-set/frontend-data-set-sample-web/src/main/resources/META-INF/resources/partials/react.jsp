<%--
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */
--%>

<%@ include file="/init.jsp" %>

<%
ReactFDSDisplayContext reactFDSDisplayContext = new ReactFDSDisplayContext(request);
%>

<react:component
	module="{SampleReactFrontendDataSet} from frontend-data-set-sample-web"
	props='<%=
		HashMapBuilder.<String, Object>put(
			"apiURL", reactFDSDisplayContext.getAPIURL()
		).put(
			"id", FDSSampleFDSNames.REACT
		).put(
			"style", "fluid"
		).put(
			"views", reactFDSDisplayContext.getViews()
		).build()
	%>'
/>