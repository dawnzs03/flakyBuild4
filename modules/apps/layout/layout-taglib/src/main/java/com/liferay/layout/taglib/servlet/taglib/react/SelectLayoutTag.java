/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.layout.taglib.servlet.taglib.react;

import com.liferay.layout.taglib.internal.servlet.ServletContextUtil;
import com.liferay.layout.taglib.internal.util.LayoutUtil;
import com.liferay.portal.kernel.json.JSONArray;
import com.liferay.portal.kernel.json.JSONFactoryUtil;
import com.liferay.portal.kernel.json.JSONUtil;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.Group;
import com.liferay.portal.kernel.model.LayoutConstants;
import com.liferay.portal.kernel.service.LayoutServiceUtil;
import com.liferay.portal.kernel.theme.ThemeDisplay;
import com.liferay.portal.kernel.util.GetterUtil;
import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.kernel.util.ParamUtil;
import com.liferay.portal.kernel.util.WebKeys;
import com.liferay.portal.util.PropsValues;
import com.liferay.taglib.util.IncludeTag;

import java.util.Map;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.jsp.PageContext;

/**
 * @author Eudaldo Alonso
 * @author Marko Cikos
 */
public class SelectLayoutTag extends IncludeTag {

	public String getItemSelectorReturnType() {
		return _itemSelectorReturnType;
	}

	public String getItemSelectorSaveEvent() {
		return _itemSelectorSaveEvent;
	}

	public String getNamespace() {
		return _namespace;
	}

	public boolean isCheckDisplayPage() {
		return _checkDisplayPage;
	}

	public boolean isEnableCurrentPage() {
		return _enableCurrentPage;
	}

	public boolean isMultiSelection() {
		return _multiSelection;
	}

	public boolean isPrivateLayout() {
		return _privateLayout;
	}

	public void setCheckDisplayPage(boolean checkDisplayPage) {
		_checkDisplayPage = checkDisplayPage;
	}

	public void setEnableCurrentPage(boolean enableCurrentPage) {
		_enableCurrentPage = enableCurrentPage;
	}

	public void setItemSelectorReturnType(String itemSelectorReturnType) {
		_itemSelectorReturnType = itemSelectorReturnType;
	}

	public void setItemSelectorSaveEvent(String itemSelectorSaveEvent) {
		_itemSelectorSaveEvent = itemSelectorSaveEvent;
	}

	public void setMultiSelection(boolean multiSelection) {
		_multiSelection = multiSelection;
	}

	public void setNamespace(String namespace) {
		_namespace = namespace;
	}

	@Override
	public void setPageContext(PageContext pageContext) {
		super.setPageContext(pageContext);

		setServletContext(ServletContextUtil.getServletContext());
	}

	public void setPrivateLayout(boolean privateLayout) {
		_privateLayout = privateLayout;
	}

	@Override
	protected void cleanUp() {
		super.cleanUp();

		_checkDisplayPage = false;
		_enableCurrentPage = false;
		_itemSelectorReturnType = null;
		_itemSelectorSaveEvent = null;
		_multiSelection = false;
		_namespace = null;
		_privateLayout = false;
	}

	@Override
	protected String getPage() {
		return _PAGE;
	}

	@Override
	protected void setAttributes(HttpServletRequest httpServletRequest) {
		try {
			httpServletRequest.setAttribute(
				"liferay-layout:select-layout:data", _getData());
		}
		catch (Exception exception) {
			_log.error(exception);
		}
	}

	private Map<String, Object> _getData() throws Exception {
		HttpServletRequest httpServletRequest = getRequest();

		ThemeDisplay themeDisplay =
			(ThemeDisplay)httpServletRequest.getAttribute(
				WebKeys.THEME_DISPLAY);

		String[] selectedLayoutIds = ParamUtil.getStringValues(
			httpServletRequest, "layoutUuid");

		return HashMapBuilder.<String, Object>put(
			"checkDisplayPage", _checkDisplayPage
		).put(
			"config",
			HashMapBuilder.<String, Object>put(
				"loadMoreItemsURL",
				themeDisplay.getPathMain() + "/portal/get_layouts"
			).put(
				"maxPageSize",
				GetterUtil.getInteger(
					PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN)
			).build()
		).put(
			"groupId", themeDisplay.getScopeGroupId()
		).put(
			"itemSelectorReturnType", _itemSelectorReturnType
		).put(
			"itemSelectorSaveEvent", _itemSelectorSaveEvent
		).put(
			"multiSelection", _multiSelection
		).put(
			"namespace", _namespace
		).put(
			"nodes", _getLayoutsJSONArray(selectedLayoutIds, themeDisplay)
		).put(
			"privateLayout", _privateLayout
		).put(
			"selectedLayoutIds", selectedLayoutIds
		).build();
	}

	private JSONArray _getLayoutsJSONArray(
			String[] selectedLayoutIds, ThemeDisplay themeDisplay)
		throws Exception {

		Group group = themeDisplay.getScopeGroup();

		if ((_privateLayout && !group.hasPrivateLayouts()) ||
			(!_privateLayout && !group.hasPublicLayouts())) {

			return JSONFactoryUtil.createJSONArray();
		}

		return JSONUtil.put(
			JSONUtil.put(
				"children",
				LayoutUtil.getLayoutsJSONArray(
					_checkDisplayPage, _enableCurrentPage,
					themeDisplay.getScopeGroupId(), getRequest(),
					_itemSelectorReturnType, _privateLayout, 0,
					selectedLayoutIds, 0,
					GetterUtil.getInteger(
						PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN))
			).put(
				"disabled", true
			).put(
				"expanded", true
			).put(
				"hasChildren", true
			).put(
				"icon", "home"
			).put(
				"id", "0"
			).put(
				"name", themeDisplay.getScopeGroupName()
			).put(
				"paginated",
				() -> {
					int layoutsCount = LayoutServiceUtil.getLayoutsCount(
						themeDisplay.getScopeGroupId(), _privateLayout,
						LayoutConstants.DEFAULT_PARENT_LAYOUT_ID);

					if (layoutsCount >
							PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN) {

						return true;
					}

					return false;
				}
			));
	}

	private static final String _PAGE = "/select_layout/page.jsp";

	private static final Log _log = LogFactoryUtil.getLog(
		SelectLayoutTag.class);

	private boolean _checkDisplayPage;
	private boolean _enableCurrentPage;
	private String _itemSelectorReturnType;
	private String _itemSelectorSaveEvent;
	private boolean _multiSelection;
	private String _namespace;
	private boolean _privateLayout;

}