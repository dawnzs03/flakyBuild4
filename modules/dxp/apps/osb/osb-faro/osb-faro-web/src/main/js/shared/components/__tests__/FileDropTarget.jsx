import React from 'react';
import {cleanup, render} from '@testing-library/react';
import {DndProvider} from 'react-dnd';
import {FileDropTarget, FileItem, getFileSizeLabel} from '../FileDropTarget';
import {HTML5Backend} from 'react-dnd-html5-backend';
import {shallow} from 'enzyme';

jest.unmock('react-dom');

describe('FileDropTarget', () => {
	describe('FileDropTarget', () => {
		afterEach(cleanup);

		it('should render', () => {
			const {container} = render(
				<DndProvider backend={HTML5Backend}>
					<FileDropTarget onChange={jest.fn()} uploadURL='/foo' />
				</DndProvider>
			);

			expect(container).toMatchSnapshot();
		});

		it('should show error if file status code is an error', () => {
			const component = shallow(
				<FileDropTarget name='fooBar' uploadURL='/foo' />
			);

			component
				.instance()
				.handleFileProgress({completed: true, status: 400});

			jest.runAllTimers();

			expect(component).toMatchSnapshot();
		});

		it('should render with file', () => {
			const component = shallow(
				<FileDropTarget
					name='fooBar'
					onChange={jest.fn()}
					uploadURL='/foo'
				/>
			);

			component.setState({
				file: {
					completed: true,
					response: 'Test.csv'
				}
			});

			jest.runAllTimers();

			expect(component).toMatchSnapshot();
		});
	});

	describe('FileItem', () => {
		afterEach(cleanup);

		it('renders', () => {
			const {container} = render(
				<DndProvider backend={HTML5Backend}>
					<FileItem file={{name: 'Test.csv'}} />
				</DndProvider>
			);

			expect(container).toMatchSnapshot();
		});

		it('renders w/ error', () => {
			const {container} = render(
				<DndProvider backend={HTML5Backend}>
					<FileItem
						file={{completed: true, name: 'Test.csv', status: 400}}
					/>
				</DndProvider>
			);

			expect(container.querySelector('.error-message')).not.toBeNull();
			expect(container.querySelector('.failure-invert')).not.toBeNull();

			expect(container.querySelector('.file-size')).toBeNull();
		});

		it('renders w/ success', () => {
			const {container} = render(
				<DndProvider backend={HTML5Backend}>
					<FileItem
						file={{completed: true, name: 'Test.csv', size: 1024}}
					/>
				</DndProvider>
			);

			expect(container.querySelector('.success-invert')).not.toBeNull();
			expect(container.querySelector('.file-size')).not.toBeNull();
		});
	});

	describe('getfilesizelabel', () => {
		it.each`
			value      | label
			${950}     | ${'0.9 KB'}
			${1024}    | ${'1 KB'}
			${1023310} | ${'999.3 KB'}
			${1048510} | ${'1 MB'}
			${7426254} | ${'7.1 MB'}
		`('should convert $value to $label', ({label, value}) => {
			expect(getFileSizeLabel(value)).toBe(label);
		});
	});
});
