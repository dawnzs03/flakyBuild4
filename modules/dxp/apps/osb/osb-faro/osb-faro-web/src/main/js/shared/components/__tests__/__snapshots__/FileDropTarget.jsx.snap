// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDropTarget FileDropTarget should render 1`] = `
<div>
  <div
    class="file-drop-target-root"
  >
    <div
      class="drop-target-container"
    >
      <button
        class="button-root btn btn-sm btn-secondary"
        type="button"
      >
        Select File
      </button>
      <div
        class="drop-zone"
      >
        Or Drag and Drop Files Here
      </div>
    </div>
  </div>
</div>
`;

exports[`FileDropTarget FileDropTarget should render with file 1`] = `
<div
  className="file-drop-target-root"
>
  <div
    className="uploaded-file"
  >
    <Component
      file={
        Object {
          "completed": true,
          "response": "Test.csv",
        }
      }
      onCancel={[Function]}
    />
  </div>
</div>
`;

exports[`FileDropTarget FileDropTarget should show error if file status code is an error 1`] = `
<div
  className="file-drop-target-root"
>
  <div
    className="uploaded-file"
  >
    <Component
      file={
        Object {
          "completed": true,
          "status": 400,
        }
      }
      onCancel={[Function]}
    />
  </div>
</div>
`;

exports[`FileDropTarget FileItem renders 1`] = `
<div>
  <div
    class="file-item"
  >
    <div
      class="d-flex flex-row align-items-center justify-content-between"
    >
      <div>
        <div
          class="file-name d-flex"
        >
          <span
            class="text-truncate"
            data-tooltip="false"
            title=""
          >
            Test.csv
          </span>
          <div
            class="status-wrapper"
          >
            <span
              aria-hidden="true"
              class="loading-root center loading-animation"
            />
          </div>
        </div>
      </div>
      <div
        class="d-flex align-items-center"
      >
        <button
          class="button-root btn btn-sm btn-link"
          type="button"
        >
          Remove
        </button>
      </div>
    </div>
  </div>
</div>
`;
